hyperspy/drawing/utils.py

```diff
@@ -878,135 +911,206 @@ def check_list_length(arg, arg_name):
 
     # Replot: create a list to store references to the images
     replot_ims = []
+    
+    #Below is for overlayed images
+    if overlay:
+        
+        #Check if images all have same scale and therefore can be overlayed.
+        for im in images:
+            if (im.axes_manager[0].scale != 
+                images[0].axes_manager[0].scale):
+                raise ValueError("Images are not the same scale and so should"
+                                 "not be overlayed.")
+        
+        original_cmap = plt.rcParams['image.cmap']
+        plt.rcParams['image.cmap'] = 'gray'
+        import matplotlib.patches as mpatches
+        ax = f.add_subplot(1, 1, 1)
+        patches = []
+        
+        #If no colors are selected use BASE_COLORS
+        if colors == []:
+            for i in range(len(images)):
+                colors.append(list(BASE_COLORS)[i])
+        
+        #If no alphas are selected use 1.0
+        if alphas == []:
+            for i in range(len(images)):
+                alphas.append(1.0)
+    
+        ax.imshow(np.zeros_like(images[0].data))
+        
+        #Loop through each image
+        for i, im in enumerate(images):
+            cols = ['k', colors[i]]
+            cmap=LinearSegmentedColormap.from_list('cmap'+str(i), cols)
+            my_cmap = cmap(np.arange(cmap.N))
+            my_cmap[:,-1] = np.linspace(0.3, 1, cmap.N)
+            my_cmap = ListedColormap(my_cmap)
```

Comment:
Is this necessary? The following seems to be doing what we need?
```python
import numpy as np
from matplotlib.colors import LinearSegmentedColormap
import matplotlib.pyplot as plt

# cmap = LinearSegmentedColormap.from_list("", ["black", "red"])
cmap = LinearSegmentedColormap.from_list("", ["white", "red"])

plt.figure()
plt.imshow(np.arange(100).reshape((10, 10)), cmap=cmap)
plt.colorbar()
```

Reply:
My code adds a transparency to the colormaps so that you can see the different images when overlayed. If you use the code as you've written above, you will only see one image at a time (although you could play with the alphas to show others).

Reply:
Sorry, I should have said it adds a transparency gradient.